enhancements/sig-architecture/70-spread-policy/README.md

```diff
@@ -0,0 +1,261 @@
+
+### Use Case 6: Spread with Ratio (Out of Scope)
+
+Sometimes, we may need to spread the workload with a specified ratio.
+Let us consider a scenario where most of the access to the workload is from the east coast of the US, even spread between us-east-1 and us-west-1 would be inappropriate.
+Instead, we want to place more workload on us-east-1 than us-west-1.
+Besides, we want to control such preferences by user-specified configurations.
+
+**Example.** We want six clusters to deploy our backend service.
+Moreover, we want 2/3 of them to be located in us-east-1 and 1/3 to be located in us-west-1 since there are more clients from the east coast of the US.
+
+## Goal
+Generally, we have two goals for the spread policy:
+1. Cover the full functionality in the Motivation section;
+2. Implement a high-performance and scalable mechanism that works for thousands of clusters efficiently.
+
+## Proposal
+
+### API Spec
+```go
+type PlacementSpec struct {
+	// ...
+
+	// SpreadPolicy defines how placement decisions should be distributed among a
+	// set of clusters.
+	// +optional
+	SpreadPolicy SpreadPolicy `json:"spreadPolicy,omitempty"`
+}
+
+// SpreadPolicy defines how placement decisions should be spreaded among the
+// clusters.
+type SpreadPolicy struct {
+	// SpreadConstraints defines how placement decision should be distributed among a
+	// set of clusters.
+	// +optional
+	// +kubebuilder:validation:MaxItems=8
+	SpreadConstraints []SpreadConstraintsTerm `json:"spreadConstraints,omitempty"`
+}
+
+// SpreadConstraintsTerm defines a terminology to spread placement decisions
+type SpreadConstraintsTerm struct {
+	// TopologyKey is either a label key or a cluster claim name of ManagedClusters
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$`
+	// +kubebuilder:validation:MaxLength=316
+	TopologyKey string `json:"topologyKey"`
+
+	// TopologyKeyType indicates the type of TopologyKey. It could be Label or Claim.
+	// +required
+	// +kubebuilder:validation:Required
+	TopologyKeyType TopologyKeyType `json:"topologyKeyType"`
+
+	// MaxSkew describes the degree to which the workload may be unevenly distributed.
+	// If present, the scheduler will not schedule if the MaxSkew cannot be satisfied.
+	// The minimum possible value is 1.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	MaxSkew *int32 `json:"maxSkew,omitempty"`
+
+	// Weight indicates the weight of the SpreadPolicyTerm. It is required only when multiple
+	// terms are used. To spread the workload, the scheduler takes the even constraints with
+	// higher weight into prior consideration.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	Weight int32 `json:"weight,omitempty"`
```

Comment:
In the current design, the spreadPolicy has 2 weight, one is the “prioritizer” weight, another is the “SpreadConstraints” [weight](https://github.com/open-cluster-management-io/enhancements/pull/70/files#diff-03ac0f150ce968ff17c6967fd1616808dea99017fe744703bc4aba9c1f97f931R156) .

Since the “SpreadConstraints” weight only impact the order/priority when selecting clusters in the [implementation](https://github.com/open-cluster-management-io/placement/compare/main...huaouo:ocm-placement:main#diff-51801281ce54e097a0a393a3b8ca7b65121c521de25bc6a70bdc725264dc944dR49) . I’m thinking if we can remove this “SpreadConstraints” weight, just use the nature order of SpreadConstraints array to select clusters. It’s easy to understand and avoid bringing too much “weight” in placement.